src/analysis/seminorm.lean

```diff
@@ -348,6 +398,19 @@ begin
   ...    < r   : by rwa mem_ball_zero at hy,
 end
 
+lemma finset_sup_ball_inter (p : ι → seminorm 𝕜 E) (s : finset ι) (r : ℝ) (hr : 0 < r):
+  ball (s.sup p) 0 r = ⋂ (i ∈ s), ball (p i) (0 : E) r :=
+begin
+  ext,
+  simp_rw [mem_Inter, mem_ball_zero],
+  refine ⟨λ hx i hi, (finset_le_sup p s i hi x).trans_lt hx, λ hx, _⟩,
```

Comment:
I don't know how much these lemmas help, but it would be good to add:
```lean
lemma ball_mono {p : seminorm 𝕜 E} {r₁ r₂ : ℝ} (h : r₁ ≤ r₂) : p.ball x r₁ ⊆ p.ball x r₂ :=
λ x (hx : _ < _), hx.trans_le h

lemma ball_antimono {p q : seminorm 𝕜 E} (h : q ≤ p) : p.ball x r ⊆ q.ball x r :=
λ x, (h _).trans_lt
```
where in principle `ball_antimono` could be use in this proof.

Reply:
`ball_mono` should be written in terms of `monotone`.

Reply:
I disagree, that uses `le` on sets which introduces mess into the goal state.

Reply:
There's precedent for this. If you're really against it, then we can instead have both `ball_mono` and `ball_subset_ball`.

Reply:
Can you point me to the precedent?

Reply:
If it is fine with you I would add these two lemmas in another PR (I need `ball_mono` there anyways).